src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -390,17 +394,22 @@ private function readProperty($zval, $property, bool $ignoreInvalidProperty = fa
 
         $result = self::$resultProto;
         $object = $zval[self::VALUE];
-        $access = $this->getReadAccessInfo(\get_class($object), $property);
+        $class = \get_class($object);
+        $access = $this->getReadAccessor($class, $property);
+
+        if (null !== $access) {
+            if (ReadAccessor::TYPE_METHOD === $access->getType()) {
+                $result[self::VALUE] = $object->{$access->getName()}();
```

Comment:
you should early return here. the sequence of `if` that are mutually exclusive is confusing. otherwise we could do a switch, that would also make sense.

Reply:
We cannot do early return as we handle zval if the passed value is a reference

Reply:
We talk about this, and we prefer to not change this here, as it may have some undefined behavior. And would be better to do it in another PR.